### ISSUE TITLE
switch Strang integration to work with (rho X, rho e)

### ISSUE THREAD

**Issue by zingale** [starkiller-astro/Microphysics#1200]
Right now Strang integrates (X, e) and SDC integrates (rho X, rho e).  We should sync them.

algebraically, since rho is constant in Strang integration, this should not affect things much (except how atol is interpreted).  But it will make code reuse between SDC and Strang much easier.

**Comment by zingale** [comment]
maybe the way to do this is to have the state always come in via `burn_t burn_state.y[]`.  But for Strang we don't need to
include the momenta or the advective terms.

**Comment by maxpkatz** [comment]
Seems to me like we should just aim to use the SDC integrator for Strang with the advective terms set to zero.

**Comment by zingale** [comment]
yes.  I think that's a good idea.  We can constexpr them out too
